src/dune_rules/virtual_rules.ml

```diff
@@ -79,57 +55,19 @@ let impl sctx ~(lib : Library.t) ~scope =
          ]
      | Some vlib ->
        let info = Lib.info vlib in
-       if not (Lib_info.virtual_ info)
-       then
-         User_error.raise
-           ~loc:lib.buildable.loc
-           [ Pp.textf
-               "Library %s isn't virtual and cannot be implemented"
-               (Lib_name.to_string implements)
-           ];
-       let+ vlib_modules, vlib_foreign_objects =
```

Comment:
This was also moved (unchanged) to `Vimpl.make`: https://github.com/ocaml/dune/pull/12392/files#diff-654c4360f33ebd2b51644332b9d26dcf9d2e39b1a2125ae4bc9f6400a7cc412cR20